docs/content/contributing/terraform/composition.md

```diff
@@ -0,0 +1,306 @@
+
+<br>
+
+---
+
+<br>
+
+## Directory and File Structure
+
+Below is the directory and file structure expected for each AVM Terraform repository/module.
+See template repo [here](https://github.com/Azure/terraform-azurerm-avm-template).
+
+- `tests/` - (for unit tests and additional tests if required - e.g. tflint etc.)
+  - `unit/` - (optional, may use further sub-directories if required)
+- `modules/` - (for sub-modules only if used)
+- `examples/` - (all examples must deploy without successfully without requiring input - these are customer facing)
+  - `defaults` - (minimum/required parameters/variables only, heavy reliance on the default values for other parameters/variables)
+  - `<other folders for examples as required>`
+- `/...` - (Module files that live in the root of module directory)
+  - `_header.md` - (required for documentation generation)
+  - `_footer.md` - (required for documentation generation)
+  - `main.tf`
+  - `locals.tf`
+  - `variables.tf`
+  - `outputs.tf`
+  - `terraform.tf`
+  - `locals.version.tf.json` - (required for telemetry, should match the upcoming release tag)
+  - `README.md` (autogenerated)
+  - `main.resource1.tf` (If a larger module you may chose to use dot notation for each resource)
+  - `locals.resource1.tf`
+
+<br>
+
+---
+
+<br>
+
+### Directory and File Structure
+
+```txt
+/ root
+│
+├───.github/
+│   ├───actions/
+│   │   ├───avmfix/
+│   │   │   └───action.yml
+│   │   ├───docs-check/
+│   │   │   └───action.yml
+│   │   ├───e2e-getexamples/
+│   │   │   └───action.yml
+│   │   ├───e2e-testexamples/
+│   │   │   └───action.yml
+│   │   ├───linting/
+│   │   │   └───action.yml
+│   │   └───version-check/
+│   │       └───action.yml
+│   ├───policies/
+│   │   ├───avmrequiredfiles.yml
+│   │   └───branchprotection.yml
+│   ├───workflows/
+│   │   ├───e2e.yml
+│   │   ├───linting.yml
+│   │   └───version-check.yml
+│   ├───CODEOWNERS
+│   └───dependabot.yml
+├───.vscode/
+│   ├───extensions.json
+│   └───settings.json
+├───examples/
+│   ├───default/
+│   │   ├───README.md
+│   │   ├───_footer.md
+│   │   ├───_header.md
+│   │   ├───main.tf
+│   │   └───variables.tf
+│   ├───.terraform-docs.yml
+│   └───README.md
+├───modules/
+│   └───README.md
+├───tests/
+│   └───README.md
+├───.gitignore
+├───.terraform-docs.yml
+├───CODE_OF_CONDUCT.md
+├───LICENSE
+├───Makefile
+├───README.md
+├───SECURITY.md
+├───SUPPORT.md
+├───_footer.md
+├───_header.md
+├───avm
+├───avm.bat
+├───locals.telemetry.tf
+├───locals.tf
+├───locals.version.tf.json
+├───main.privateendpoint.tf
+├───main.telemetry.tf
+├───main.tf
+├───outputs.tf
+├───terraform.tf
+└───variables.tf
+```
+
+<br>
+
+---
+
+<br>
+
+## Code Styling
+
+This section points to conventions to be followed when developing a module.
+
+<br>
+
+---
+
+<br>
+
+### Casing
+
+Use `snake_casing` as per [TFNFR3](/Azure-Verified-Modules/specs/terraform/#id-tfnfr4---category-composition---code-styling---lower-snake_casing).
```

Comment:
Do we need to repeat the specs in here? Seems like double effort.

Reply:
So not repeating the specs or completely removing the whole `### Casing` section since it's mentioned in the specs already?